### PR TITLE
Docs: update visualization naming conventions 

### LINKED ISSUE #74232: Docs: Standardize visualization names
Standardize the way we write about visualizations within the Visualizations docs. Going forward visualization types will be treated as common nouns. This means:

- No capitalization
- Require an article like "a" or "an" (unless plural)
- Mostly used in the plural form (like the way we talk about dashboards, panels, data sources etc)

The exceptions are visualizations that are already in plural form or have another more common meaning. These will be:

- Written as adjectives
- They modify the word "visualization" (not "panel")

For example:

- time series visualization
- traces visualization

These guidelines will also be added in the Writer's Toolkit as part of a visualization docs template for future writing.

### PR DESCRIPTION
**Pull request by imatwawana**
Updating visualization docs with following changes:

- Make visualization names common nouns rather than proper nouns (lower case, add an article)
- Replace adjective form of visualization with noun form
- Replace the word "panel" with "visualization" where appropriate
- Fix titles to match with name of visualization in the UI

This is one of two PRs doing this work. This PR covers the visualizations for which the new naming convention isn't a question. A second PR will address trickier visualization names and references on the Visualizations index page.

**Note for reviewers**: I only need you to read the first sentence of each file and ensure that the new way we refer to the visualization doesn't sound confusing or just weird. If you want to do a closer review, feel free, but I'm just trying to ensure we don't make anything less clear.

### CODE REVIEW

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 1]
```diff
@@ -38,9 +38,9 @@ weight: 100
 
 # Geomap
 
-The Geomap panel visualization allows you to view and customize the world map using geospatial data. You can configure various overlay styles and map view settings to easily focus on the important location-based characteristics of the data.
+Geomaps allow you to view and customize the world map using geospatial data. You can configure various overlay styles and map view settings to easily focus on the important location-based characteristics of the data.
 
-> We would love your feedback on Geomap. Please check out the [Github discussion](https://github.com/grafana/grafana/discussions/62159) and join the conversation.
+> We would love your feedback on geomaps. Please check out the [Github discussion](https://github.com/grafana/grafana/discussions/62159) and join the conversation.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:43]: Can we keep this singular?

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:43]: Just in this particular instance or in general? And why?

Comment (nmarrs) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:43]: This does sound a little funny but I think we can get used to the plural version 🤷‍♂️

**Review by nmarrs — APPROVED** [review]
LGTM overall! a few nits / small points but I'm good with merging with or without addressing the nits :)

File: docs/sources/panels-visualizations/visualizations/canvas/index.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visu
 
 If multiple elements use the same field name, and you want to control which elements display the data link, you can create a unique field name using the [add field from calculation transform](https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/#add-field-from-calculation). The alias you create in the transformation will appear as a field you can use with an element.
 
-1. In the panel editor for the canvas panel, click the **Transform** tab.
+1. In the panel editor for the canvas, click the **Transform** tab.
```
[the remark this answers is not in the document]
Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/canvas/index.md:109]: I think of the panel editor as including settings that are panel level and visualization level, so I think we should leave this one.

File: docs/sources/panels-visualizations/visualizations/geomap/index.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:43]: I suspect this one just may be that we're accustomed to treating it like a brand name so it's a shift to stop doing that. But grammatically, it's solid, so I think we should stick with the our new convention here.

**Pull request merged**